client/src/actions/cartActions.js

```diff
@@ -0,0 +1,22 @@
+import {
+  CART_SAVE_SHIPPING_ADDRESS,
+  CART_SAVE_BOOKING_DATE,
+} from "../constants/cartConstants";
+// export const login = async (dispatch, loginPayload) => {
+export const saveShippingAddress = async (dispatch, data) => {
+  dispatch({
+    type: CART_SAVE_SHIPPING_ADDRESS,
+    payload: data,
+  });
+
+  localStorage.setItem("shippingAddress", JSON.stringify(data));

```

Comment:
Would it be possible to use state/context rather than storing this in LS? Also is the shipping address something that will eventually be stored in the database?

Reply:
Yes, I do have it saved inside the state. LS is used to have it persistent with the state. And the shipping address will be saved inside our Order database

Reply:
I agree with Nellie, I don't think it should be necessary to store this in Local Storage, if you have it in the DB and in state you won't need it in Local Storage

Reply:
I will do the necessary changes. The thought process behind putting it in LS was if the user decides to go back and add more items to the cart after putting his booking details then he/she does not have to go through the entire process of adding the details again inside checkout page again since state is not persistent. I will remove LS and update it.

Reply:
@Vaibhav-Kambli , I guess we can add address and other cart related info into the cart field that I worked on. We only need a route to change them-it could be a single route both for adding and editing- and that's it I guess? maybe we will need new actions but as far as backend, I already have a route that returns all the info in the cart.

my branch has been approved, after our standup I can merge that and you can add your fields into my cart context? and add actions to save the info to backend. Or I can do that.

 If we could write separate controllers for only those fields that would be ideal, because the edit controller that I have right now is already pretty big function. you can see that in my branch.  Maybe it's best to have routes only for manipulating address and all the fields you were working on.

Reply:
@MinaKhamesi I think the changes I have done now should be fine. Because the payment page is already huge since it has stripe functionality and stepper, I cannot really separate the stripe component into it's own file since the checkout button is connected to my stripe form. Also, if I add  functionality to get this data from the backend then it will get even bigger. For now this is working properly. In future we can have it refactored if we move stripe to it's own component